src/app/rundown-view/components/follow-playhead-timeline/follow-playhead-timeline.component.ts

```diff
@@ -77,6 +86,19 @@ export class FollowPlayheadTimelineComponent implements OnChanges {
     }
   }
 
+  private checkForSpanningElements(parts: readonly Part[]): Part[] {
+    return parts.reduce((result: Part[], currentPart: Part, index: number) => {
+      const previousPart = result[index - 1]
+      const prevPieceWithSpanElement = previousPart?.pieces.findIndex(piece => piece.lifespan === PieceLifespan.SPANNING_UNTIL_SEGMENT_END)
+      const currPieceHasSpanElement = currentPart.pieces.some(piece => piece.lifespan === PieceLifespan.SPANNING_UNTIL_SEGMENT_END)
+
+      if (prevPieceWithSpanElement >= 0 && !currPieceHasSpanElement) {
+        currentPart.pieces.push({ ...previousPart.pieces[prevPieceWithSpanElement], isSpanned: true } as Piece)
```

Comment:
You should avoid the `as` keyword as much as possible in production code. You are basically telling Typescript to avoid using types.